tests/series/test_size_bytes.py

```diff
@@ -195,3 +196,36 @@ def test_series_fixed_size_list_size_bytes(dtype, size) -> None:
 
     assert s.datatype() == DataType.from_arrow_type(list_dtype)
     assert s.size_bytes() == data.nbytes
+
+
+@pytest.mark.skipif(
+    not PYARROW_GE_7_0_0,
+    reason="Array.nbytes behavior changed in versions >= 7.0.0. Old behavior is incompatible with our tests and is renamed to Array.get_total_buffer_size()",
+)
+@pytest.mark.parametrize("size", [0, 1, 2, 8, 9, 16])
+def test_series_struct_size_bytes(size) -> None:
+    dtype1, dtype2, dtype3 = pa.int64(), pa.float64(), pa.string()
+    dtype = pa.struct({"a": dtype1, "b": dtype2, "c": dtype3})
+    pydata = [
+        {"a": 3 * i, "b": 3 * i + 1, "c": 3 * i + 2} if i % 2 == 0 else {"a": 3 * i, "c": 3 * i + 2}
+        for i in range(size)
+    ]
+    data = pa.array(pydata).cast(dtype)
+
+    s = Series.from_arrow(data)
+
+    assert s.datatype() == DataType.from_arrow_type(dtype)
+    # TODO(Clark): Investigate this discrepancy.
+    # TODO(Clark): Investigate this discrepancy between Arrow2 and pyarrow.
+    # We should fix this upstream and/or refactor these tests to not rely on pyarrow as the source-of-truth.
+    assert s.size_bytes() == data.nbytes + 8 + (size * 8) // 2
+
+    ## with nulls
+    if size > 0:
+        pydata = pydata[:-1] + [None]
+    data = pa.array(pydata).cast(dtype)
+
+    s = Series.from_arrow(data)
+
+    assert s.datatype() == DataType.from_arrow_type(dtype)
+    assert s.size_bytes() == data.nbytes + 8 + (size * 8) // 2 + math.ceil(size / 8)
```

Comment:
Still need to look into this discrepancy... I'm sure this is partially the off-by-one error for the underlying string array offsets, and partially some nested type discrepancy similar to the one we've already detected around variable-length list arrays.

Reply:
Arrow spec says that offsets should be length+1 (https://arrow.apache.org/docs/format/Columnar.html#variable-size-binary-layout)

I think our logic might be off by 1

Reply:
@samster25 said that pyarrow's `str_array.nbytes` has an off-by-one error due to their offset array accounting, which arrow2 does not have, so I think that our logic (which delegates to Arrow2) is fine there.

That would explain the `+ 8` difference, but I haven't looked into why the `+ (size * 8) // 2 + math.ceiling(size / 8)` is required to get arrow2 and pyarrow to agree.